---
title: "Overview of methods for differential splicing and sQTL analyses"
date: "24 May 2016"
author: "Malgorzata Nowicka"
output: html_document
bibliography: refs.bib
csl: f1000research.csl
---

The source code of this file can be found on [GitHub](https://github.com/gosianow/DS_methods).

| Application | DS method | Counting method | Type of DS analysis| Modeling| Description of the DS method| Extra notes/features|
|:-----|:-----|:-----|:-----|:-------|------------------------|--------------|
|DS| [DEXSeq](http://bioconductor.org/packages/release/bioc/html/DEXSeq.html) [@Anders2012] | [DEXSeq](http://bioconductor.org/packages/release/bioc/html/DEXSeq.html)/ [HTSeq](http://www-huber.embl.de/HTSeq/doc/overview.html) [@Anders2012; @Anders2015]| Local; exon usage| At the gene level but exons are treated as independent| Fits, per gene, a generalized linear model with bin-condition interactions assuming the negative-binomial distribution for counts; Each exon-condition interaction is fitted within a separate model containing interaction coefficient for only this exon; Fits sample effects to handle the gene expression variability; Involves sharing information between genes, for the dispersion estimation| Allows analyses with multiple covariates; Returns exon-level p-values, but has a module to transform these p-values into gene-level adjusted p-values|
|sQTL| [Altrans](https://sourceforge.net/projects/altrans/)/ [FastQTL](http://fastqtl.sourceforge.net/) [@Ongen2015Altrans; @Ongen2015] | [Altrans](https://sourceforge.net/projects/altrans/) [@Ongen2015Altrans] | Local; exon link (junction) usage| Modeling each exon link separately| For the exon-link quantifications, it uses the paired-end reads with mates mapping to different exons and junction reads, and account for the insert size; Inference based on the F value which is a ratio between the coverage of a link and the sum of coverages of all the links that the primary exon makes;  Employs Pearson correlation, from [FastQTL](http://fastqtl.sourceforge.net/), as a measure of association between genotype and Gaussian transformed exon link ratios | Applies a permutation approach to estimate p-values|
|DS| [rMATS](http://rnaseq-mats.sourceforge.net/) [@Shen2014] | [rMATS](http://rnaseq-mats.sourceforge.net/)| Local; splicing events| Modeling each splicing event separately| For the splicing event quantification, it uses the inclusion reads and exclusion reads (no reads that map to the body of constitutive exons); Uses a hierarchical model: binomial distribution for the inclusion read counts given the inclusion level PSI to model the estimation uncertainty of PSI in individual replicate and logit-normal mixed model  for PSI with random effects for each sample to account for the variability among replicates within conditions | Allows unpaired and paired comparisons|
|sQTL| [GLiMMPS](http://www.mimg.ucla.edu/faculty/xing/glimmps/) [@Zhao2013] | [GLiMMPS](http://www.mimg.ucla.edu/faculty/xing/glimmps/)| Local; splicing events| Modeling each splicing event separately| For the splicing event quantification, it uses the inclusion reads and exclusion reads (no reads that map to the body of constitutive exons); Uses a GLMM with a hierarchical model: binomial distribution for the inclusion read counts given the inclusion level PSI to model the estimation uncertainty of PSI in individual replicate and logit-normal mixed model  for PSI with fixed effects for the genotype and random effects for each sample to account for the variability within the same genotype group; | Computation done with glmer function from R package lme4; Using permutations for the FDR estimation|
|sQTL| Jia et al. [@Jia2014] | [PennSeq](https://sourceforge.net/projects/pennseq/) [@Hu2014] | Local; splicing events| Modeling each splicing event separately| For the splicing event quantification, PennSeq uses all reads mapped to a given exon-trio, i.e.,  the  inclusion reads and exclusion reads and reads that map to the body of the constitutive exons, and it accounts for the paired-end nature of the data; Applies  a random-effects meta regression (REMR) using logit-normal for inclusion levels PSI with fixed effects for the genotype, random effects for each sample and a component representing the PSI estimation uncertainty from PennSeq; | Computation is performed with metafor R package; Jia et al. considered also a beta regression and GLMM like in [GLiMMPS](http://www.mimg.ucla.edu/faculty/xing/glimmps/) but REMR was performing the best|
DS| [MISO](http://genes.mit.edu/burgelab/miso/) [@Katz2010] | [MISO](http://genes.mit.edu/burgelab/miso/)| Local; splicing events; Global; transcript usage| Modeling each splicing event separately in exon-centric analyses; Multivariate for isoform-centric analyses| For the quantification of exon or isoform inclusion levels, it employs a Bayesian inference which uses  the  inclusion reads and exclusion reads and reads that map to the body of the constitutive exons and captures the information about library inserts in paired-end data; The inference about DS between two samples is based on the  posterior probabilities of isoform inclusions  (or exon-inclusions) which are compared using the Bayes factor| DS analysis only for 1 vs. 1 comparisons; Performs exon-centric and isoform-centric DS analyses|
|DS| [Cuffdiff2](http://cole-trapnell-lab.github.io/cufflinks/) [@Trapnell2013] | [Cufflinks](http://cole-trapnell-lab.github.io/cufflinks/) [@Trapnell2010]| Global; transcript usage | Multivariate| Uses the Jensen-Shannon divergence metric on probability distributions of isoform proportions (obtained from Cufflinks) as a measure of changes in isoform relative abundances between samples;  The test statistic is the square root of the Jensen-Shannon divergence divided by its standard error (one-sided t -test)| DS based on transcripts grouped by TSS or promoter usage|
|sQTL| [sQTLseekeR](https://github.com/jmonlong/sQTLseekeR) [@Monlong2014] | | Global; transcript usage| Multivariate| To account for the gene expression, it transforms transcript quantifications into ratios; To test for the association between a genotype and transcript ratios, it uses a test proposed by Anderson which is similar to a multivariate analysis of variance (MANOVA) without assuming any probabilistic distribution. The difference between the within-group and between-group variability is measured by a pseudo-F ratio score; The Hellinger distance in used as a dissimilarity measure between transcript ratios| Employs permutations to estimate p-values; Allows comparison between multiple groups of samples|
|DS | [LeafCutter](https://github.com/davidaknowles/leafcutter) [@Li2016] | [LeafCutter](https://github.com/davidaknowles/leafcutter) | Local; intron usage| Multivariate | For the quantification of intron excision, uses only the junction reads; For the DS inference, uses the Dirichlet-multinomial generalized linear model | Allows analyses with multiple covariates; Annotation free|
|sQTL | [LeafCutter](https://github.com/davidaknowles/leafcutter)/ [FastQTL](http://fastqtl.sourceforge.net/) [@Li2016; @Ongen2015] | [LeafCutter](https://github.com/davidaknowles/leafcutter) [@Li2016] | Local; intron usage | Modeling each intron separately | For the quantification of intron excision, uses only the junction reads; Inference based on the proportions of reads supporting excised introns; Employs Pearson correlation, from [FastQTL](http://fastqtl.sourceforge.net/) [@Ongen2015], as a measure of association between genotype and intron excision proportions | Applies a permutation approach to estimate p-values; Annotation free|
|DS and sQTL | [DRIMSeq](https://www.bioconductor.org/packages/3.3/bioc/html/DRIMSeq.html) | | Global; transcript usage| Multivariate| Uses the Dirichlet-multinomial distribution to model transcript counts of a gene in each condition and for the pooled data; To test for the differences in transcript proportions between conditions, DRIMSeq uses the LR statistic; Involves sharing information between genes, for the dispersion estimation| Allows comparison between multiple groups of samples; In sQTL analysis, a permutation approach is employed in p-value estimation to  account for the dependencies between variants (SNPs)|
|sQTL| Lappalainen et al. (GEUVADIS) [@Lappalainen2013] | [FluxCapacitor](http://sammeth.net/confluence/display/FLUX/Home) [@Montgomery2010]| Global; transcript usage| Modeling each transcript separately| To account for the gene expression, they transformed transcript quantifications into ratios; They used a linear model implemented in Matrix eQTL to test each transcript separately for the association between its expression and genotype | FDR was estimated by permutations |
|sQTL| Montgomety et al. [@Montgomery2010] | [FluxCapacitor](http://sammeth.net/confluence/display/FLUX/Home) [@Montgomery2010] | Local; splicing events| Modeling each splicing event separately| Association analyses between genotypes and splicing events conducted with Spearman rank correlation| P-value significance evaluated with permutations|
|sQTL| Battle et al. [@Battle2014]| [Cufflinks](http://cole-trapnell-lab.github.io/cufflinks/) [@Trapnell2010] | Global; transcript usage| Modeling each transcript separately| To account for the gene expression, it transforms transcript quantifications   into ratios; Association analyses between genotypes and transcript ratios conducted with Spearman rank correlation| Bonferroni correction used  to account for the number of SNPs tested per gene|
|sQTL| Pickrell et al. [@Pickrell2010]| | Local; exon usage| Modeling each exon separately| To account for the gene expression, it transforms exon quantifications into ratios; Association analyses between genotypes and splicing events were done with standard linear regressions between exon quantifications and genotypes| Using permutations for the FDR estimation|


# Other overviews 

Alamancos et al. [@Alamancos2014] provide a comprehensive overview of methods to study splicing from RNA-seq, which perform mapping, reconstruction of isoforms or splicing events, quantification and comparison. 

Hooper et al. [@Hooper2014] investigate the performance of DS methods.

Soneson et al. [@Soneson2016] compare different types of quantifications (e.g., exon counts, transcript abundances) to conduct DS analyses. They developed a simulation framework that mimic real RNA-seq data and can be used for comparison of other DS methods. 

Kanitz et al. [@Kanitz2015] and Teng et al. [@Teng2016] provide an overview of methods for transcript quantification and investigate their performance. Additionally, Teng et al. developed a webtool where other methods can be submitted for comparison.



# References














