# Overview of methods for differential splicing and sQTL analyses


| Application | DS method | Counting method | Type of DS analysis| Modeling| Description of the DS method| Extra notes/features|
|:-----|:-----|:-----|:-----|:-------|------------------------|--------------|
|DS| voom-diffSplice (limma) | featureCounts (Rsubread) |Local; exon usage | At the gene level but exons are treated as independent| Fits, per gene, a normal weighted linear model with exon-condition interactions to voom-transformed counts (log-counts per million); Voom observational weights are inversely proportional to count variance and account for the mean-variance relationship within RNA-seq data; All exon-condition interactions are fitted at once; Fits sample effects to handle the gene expression variability; Involves sharing information between genes, for the dispersion estimation| Allows analysis with multiple covariates which can be continuous or categorical; Returns gene-level p-values|
|DS| DEXSeq | DEXSeq/ HTSeq | Local; exon usage| At the gene level but exons are treated as independent| Fits, per gene, a generalized linear model with bin-condition interactions assuming the negative-binomial distribution for counts; Each exon-condition interaction is fitted within a separate model containing interaction coefficient for only this exon; Fits sample effects to handle the gene expression variability; Involves sharing information between genes, for the dispersion estimation| Allows analyses with multiple covariates; Returns exon-level p-values, but has a module to transform these p-values into gene-level adjusted p-values|
|sQTL| Altrans/ FastQTL | Altrans | Local; exon link (junction) usage| Modeling each exon link separately| For the exon-link quantifications, it uses the paired-end reads with mates mapping to different exons and junction reads, and account for the insert size; Inference based on the F value which is a ratio between the coverage of a link and the sum of coverages of all the links that the primary exon makes;  Employs Pearson correlation, from FastQTL, as a measure of association between genotype and Gaussian transformed exon link ratios | Applies a permutation approach to account for the dependencies among the link quantifications of a gene and between the variants within the cis-window of that gene in order to detect the significant asQTL genes|
|DS| rMATS  | rMATS| Local; splicing events| Modeling each splicing event separately| For the splicing event quantification, it uses the inclusion reads and exclusion reads (no reads that map to the body of constitutive exons); Uses a hierarchical model: binomial distribution for the inclusion read counts given the inclusion level PSI to model the estimation uncertainty of PSI in individual replicate and logit-normal mixed model  for PSI with random effects for each sample to account for the variability among replicates within conditions | Allows unpaired and paired comparisons|
|sQTL| GLiMMPS  | GLiMMPS| Local; splicing events| Modeling each splicing event separately| For the splicing event quantification, it uses the inclusion reads and exclusion reads (no reads that map to the body of constitutive exons); Uses a GLMM with a hierarchical model: binomial distribution for the inclusion read counts given the inclusion level PSI to model the estimation uncertainty of PSI in individual replicate and logit-normal mixed model  for PSI with fixed effects for the genotype and random effects for each sample to account for the variability within the same genotype group; | Computation done with glmer function from R package lme4; Using permutations to estimated the FDR|
|sQTL| Jia et al.  | PennSeq | Local; splicing events| Modeling each splicing event separately| For the splicing event quantification, PennSeq uses all reads mapped to a given exon-trio, i.e.,  the  inclusion reads and exclusion reads and reads that map to the body of the constitutive exons, and it accounts for the paired-end nature of the data; Applies  a random-effects meta regression using logit-normal for inclusion levels PSI with fixed effects for the genotype, random effects for each sample and a component representing the PSI estimation uncertainty from PennSeq; | Computation is performed with metafor R package; Jia et al. considered also a beta regression and GLMM like in GliMMPS but REMR was performing the best|
DS| MISO | MISO| Local; splicing events; Global; transcript usage| Modeling each splicing event separately in exon-centric analyses; Multivariate for isoform-centric analyses| For the quantification of exon or isoform inclusion levels, it employs a Bayesian inference which uses  the  inclusion reads and exclusion reads and reads that map to the body of the constitutive exons and captures the information about library inserts in paired-end data; The inference about DS between two samples is based on the  posterior probabilities of isoform inclusions  (or exon-inclusions) which are compared using the Bayes factor| DS analysis only for 1 vs. 1 comparisons; Performs exon-centric and isoform-centric DS analyses|
|DS| Cuffdiff2  | Cufflinks | Global; transcript usage | Multivariate| Uses the Jensen-Shannon divergence metric on probability distributions of isoform proportions (obtained from Cufflinks) as a measure of changes in isoform relative abundances between samples;  The test statistic is the square root of the Jensen-Shannon divergence divided by its standard error (one-sided t -test)| DS based on transcripts grouped by TSS or promoter usage|
|sQTL| sQTLseekeR  | | Global; transcript usage| Multivariate| To account for the gene expression, it transforms transcript quantifications into ratios; To test for the association between a genotype and transcript ratios, it uses a test proposed by Anderson which is similar to a multivariate analysis of variance (MANOVA) without assuming any probabilistic distribution. The difference between the within-group and between-group variability is measured by a pseudo-F ratio score; The Hellinger distance in used as a dissimilarity measure between transcript ratios| Allows comparison between multiple groups of samples|
|DS and sQTL | DRIMSeq | | Global; transcript usage| Multivariate| Uses the Dirichlet-multinomial distribution to model transcript counts of a gene in each condition and for the pooled data; To test for the differences in transcript proportions between conditions, DRIMSeq uses the LR statistic; Involves sharing information between genes, for the dispersion estimation| Allows comparison between multiple groups of samples; In sQTL analysis, a permutation approach is employed to  account for the dependencies between variants (SNPs)|
|sQTL| Lappalainen et al. (GEUVADIS) | FluxCapacitor | Global; transcript usage| Modeling each transcript separately| To account for the gene expression, they transformed transcript quantifications into ratios; They used a linear model implemented in Matrix eQTL to test each transcript separately for the association between its expression and genotype | FDR was estimated by permutations |
|sQTL| Montgomety et al.  | FluxCapacitor  | Local; splicing events| Modeling each splicing event separately| Association analyses between genotypes and splicing events conducted with Spearman rank correlation| P-value significance evaluated with permutations|
|sQTL| Battle et al. | Cufflinks  | Global; transcript usage| Modeling each transcript separately| To account for the gene expression, it transforms transcript quantifications   into ratios; Association analyses between genotypes and transcript ratios conducted with Spearman rank correlation| Bonferroni correction used  to account for the number of SNPs tested per gene|
|sQTL| Pickrell et al. | | Local; exon usage| Modeling each exon separately| To account for the gene expression, it transforms exon quantifications into ratios; Association analyses between genotypes and splicing events were done with standard linear regressions between exon quantifications and genotypes| Employed permutations to estimate the FDR|











